### ISSUE TITLE
PasswordInput visibility toggle lacking aria-pressed attribute

### ISSUE THREAD

**Issue by astolman** [mantinedev/mantine#8424]
### Dependencies check up

- [x] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

8.3.6

### What package has an issue?

@mantine/core

### What framework do you use?

Next.js

### In which browsers you can reproduce the issue?

All

### Describe the bug

The password visibility toggle should get an aria-pressed attribute automatically applied to it. Otherwise, to implement it myself, I end up having to switch to fully control the visibility state. This seems like overkill for a simple attribute that could easily be handled internally. In my app, I'd be fine with the <PasswordInput/> with only the label prop defined, but I'm having to pass onVisibilityChange, visibilityToggleButtonProps and define a visibility state in the parent component just to have the correct aria-pressed prop.

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/hq9tr4

### Possible fix

Add aria-pressed to the visibility toggle icon.

### Self-service

- [ ] I would be willing to implement a fix for this issue
